### ISSUE TITLE
"yield from" broken

### ISSUE THREAD

**Issue by letmaik** [microsoft/pyright#619]
**Describe the bug**
> Argument of type "list[int]" cannot be assigned to parameter "k" of type "int"
  "list[int]" is incompatible with "int"

**To Reproduce**
See code below.

**Expected behavior**
It should use the item type, not the container type, when using `yield from`.

**Screenshots or Code**
```py
from typing import Dict
def g():
    yield from [1,2,3]

a = Dict[int,int]()
for i in g():
    a[i] = i # type error here: Argument of type "list[int]" cannot be assigned to parameter "k" of type "int"
```

**VS Code extension or command-line**
VS Code extension.

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. I've repro'ed the problem, and I'll work on a fix.

**Comment by erictraut** [comment]
The fix for this issue will be in the next published version of pyright. Thanks again for the bug report!

**Comment by erictraut** [comment]
This is now fixed in version 1.1.32, which I just published.
